auth-api/src/main/java/io/stargate/auth/api/resources/AuthResource.java

```diff
@@ -48,9 +47,17 @@
 
   private static final Logger logger = LoggerFactory.getLogger(AuthResource.class);
 
-  @Inject private AuthenticationService authService;
-  private static final boolean shouldEnableUsernameToken =
-      Boolean.parseBoolean(System.getProperty("stargate.auth_api_enable_username_token", "false"));
+  private final AuthenticationService authService;
+  private final boolean shouldEnableUsernameToken;
+
+  public AuthResource(AuthenticationService authService) {
```

Comment:
When resolving the conflicts I was not sure how to deal with the `@Inject` here @dougwettlaufer. I'd prefer to have one autowired constructor and inject all arguments there, including the boolean. But I guess we'd need a way to DI the boolean then.

Does this change break the code?

Reply:
As long as your tests still pass you should be fine. Although this might need an `@Inject` on the constructor?

Reply:
I'm not actually sure how it works with Java EE CDI. Which constructor will be called when the bean is registered with `environment.jersey().register(AuthResource.class);`? I guess the smart way to go about this is to use `dropwizard-configuration` and inject a specific configuration rather than a boolean. This also allows to use environment variables or config files in addition to system properties in order to specify configuration.

Is this in use already / do you think this makes sense? I fear that my change here might have broken the capability to enable the token via username functionality.

See "Configuration" under https://www.dropwizard.io/en/release-0.9.x/manual/core.html for reference.